keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/PersistentEvent.kt

```diff
@@ -19,7 +19,7 @@ import java.time.Instant
 abstract class PersistentEvent {
   abstract val scope: Scope
   abstract val application: String
-  abstract val uid: String // The unique ID of the thing associated with the scope. Defined in sub-classes.
+  abstract val ref: String // The unique ID of the thing associated with the scope. Defined in sub-classes.
```

Comment:
This will break deserialization for existing events in the DB, right? Are you planning to write a migration, or add code to the functions that retrieve events to account for that?

One trick I've used in the past, which I know is not great in terms of performance, but is simple and works, is to do something like this in those methods:
```kotlin
objectMapper.readValue<Map<String, Any?>>(json)
  .toMutableMap()
  .let {
    it.putIfAbsent("ref", it["uid"])
    objectMapper.convertValue(it, PersistentEvent::class.java)
  }
```

Reply:
Argh, yes. I need to migrate that